demisto_sdk/commands/test_content/xsoar_tools/xsoar_client.py

```diff
@@ -0,0 +1,697 @@
+
+        return raw_response
+
+    @retry_http_request()
+    def list_indicators(
+        self,
+        page: int = 0,
+        size: int = 50,
+        query: str = "",
+        response_type: str = "object",
+    ):
+        body = {"page": page, "size": size, "query": query}
+        raw_response, _, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="POST",
+            path="/indicators/search",
+            body=body,
+            response_type=response_type,
+        )
+        return raw_response
+
+    @retry_http_request()
+    def get_indicators_whitelist(self, response_type: str = "object"):
+        raw_response, _, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="GET",
+            path="/indicators/whitelisted",
+            response_type=response_type,
+        )
+        return raw_response
+
+    @retry_http_request()
+    def delete_indicators_from_whitelist(
+        self, indicator_ids: List[str], response_type: str = "object"
+    ):
+        raw_response, _, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="POST",
+            path="/indicators/whitelist/remove",
+            response_type=response_type,
+        )
+        indicator_ids = set(indicator_ids)
+        raw_response = set(raw_response)
+
+        if not indicator_ids.issubset(raw_response):
+            logger.warning(
+                f"Could not delete indicators with the following IDs: {indicator_ids.difference(raw_response)}"
+            )
+        return raw_response
+
+    @retry_http_request()
+    def get_integrations_module_configuration(
+        self, _id: Optional[str] = None, response_type: str = "object"
+    ):
+        raw_response, _, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="POST",
+            path="/settings/integration/search",
+            response_type=response_type,
+        )
+        if not _id:
+            return raw_response
+        for config in raw_response.get("configurations") or []:
+            if config.get("id") == _id:
+                return config
+
+        raise ValueError(
+            f"Could not find module configuration for integration ID '{_id}'"
+        )
+
+    @retry_http_request()
+    def get_installed_packs(self, response_type: str = "object"):
+        raw_response, _, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="GET",
+            path="/contentpacks/metadata/installed",
+            response_type=response_type,
+        )
+        return raw_response
+
+    def get_installed_pack(self, pack_id: str) -> Dict[str, Any]:
+        for pack_info in self.get_installed_packs():
+            if pack_info.get("id") == pack_id:
+                return pack_info
+        raise ValueError(f"pack ID {pack_id} does not exist in {self.base_url}")
+
+    @retry_http_request(times=20)
```

Comment:
why 20 times?

Reply:
because it might take for the long running integration a few seconds to get up.